packages/engine/src/simulation/package/context/packages/agent_messages/mod.rs

```diff
@@ -80,19 +80,19 @@ impl GetWorkerSimStartMsg for AgentMessages {
 impl Package for AgentMessages {
     async fn run<'s>(
         &mut self,
-        state: Arc<State>,
+        state: StateReadProxy,
```

Comment:
I wonder if we should rename all `state` parameters that are actually proxies to something like `state_proxy`. 

I'm in two minds, it does make the code a fair bit longer but I think it's good to distinguish between the proxies and the actual `State` object.